runner/environment.go

```diff
@@ -0,0 +1,27 @@
+package runner
+
+import (
+	"bufio"
+	"os"
+	"os/exec"
+)
+
+func (e *Engine) modifyEnvironment(c *exec.Cmd) {
+	if e.config.Build.EnvironmentFile == "" {
+		return
+	}
+
+	env := os.Environ()
+
+	envFile, err := os.Open(e.config.Build.EnvironmentFile)
+	if err != nil {
+		return
+	}
+
+	scan := bufio.NewScanner(envFile)
+	for scan.Scan() {
+		env = append(env, scan.Text())
```

Comment:
Won't you import empty line or line with comment?

Is there any library you could have used?

Reply:
Empty lines, like all lines without `=` are ignored, lines like `# this is a comment = true` add a very strange variable to the environment. Neither should pose much of a problem.

Reply:
Yes, they should be ignored. My point is what is the code that is not in this PR, that skip them

Reply:
Sorry, GitHub's lovely handling of email replies put this in the main thread.

https://github.com/air-verse/air/pull/598#issuecomment-2143825842

Otherwise it would be a simple if-statement. Still no need to import anything :) Let me know if I should add that clause.

Reply:
Yes,i was expecting an if in the scan look

Maybe something like

```go
line = strings.Split(line, "#)
line = strings.TrimSpace(line)

if line == "" {
     continue
}
```

Reply:
> Sorry, GitHub's lovely handling of email replies put this in the main thread.
> 
>


> OS handling of environment variables and Go stdlib.

 That's my point. You are not using Go stdlib when handling the variables the way you do in this PR

> Otherwise it would be a simple if-statement. Still no need to import anything :) Let me know if I should add that clause.

Yes,i was expecting an if in the scan loop

Maybe something like

```go
line = strings.Split(line, "#)
line = strings.TrimSpace(line)

if line == "" {
     continue
}
```

Reply:
I mean you are using os.Environ. so yes, it skip the comments, empty lines. 

But your code doesn't when you add the lines of the env file

Reply:
I'm worried that you would have to code everything needed to do things like this

```env
BAR=42
FOO=BAR
```

Your code will set too to BAR not 42

I'm also worried that you would have to comply with uncommon but yet still used emv syntax like this

```env

FOO=${BAR:-42} # If BAR not set or null, use 42.
```

I might be totally wrong, but from my perspective this should be at least covered by tests

Reply:
@mpldr @xiantang

Reply:
I found backfound the project I was looking for

https://github.com/joho/godotenv

I think you should reluy on a lib like this to load and parse the env file, then you can append to existing env

Reply:
> I'm worried that you would have to code everything needed to do things like this
> 
> ```dotenv
> BAR=42
> FOO=BAR
> ```
> 
> Your code will set too to BAR not 42

That's the expected behaviour, but you're right, it doesn't account for any kind of shell interpretation. I'll have a look at the suggested library later today. :)